libs/api/src/main/java/de/cyface/api/ClassifiedSegmentRetriever.java

```diff
@@ -18,29 +18,32 @@
  */
 package de.cyface.api;
 
-import java.time.OffsetDateTime;
 import java.time.ZonedDateTime;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
 
+import de.cyface.api.model.ClassifiedMeasurementSegment;
 import de.cyface.api.model.ClassifiedSegment;
-import de.cyface.api.model.Geometry;
-import de.cyface.model.Modality;
+import de.cyface.api.model.ClassifiedSegmentFactoryProvider;
+import de.cyface.api.model.User;
 import io.vertx.core.Future;
 import io.vertx.core.Promise;
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
 import io.vertx.ext.mongo.MongoClient;
 
+import static de.cyface.api.model.ClassifiedSegmentFactoryProvider.Mode.MEASUREMENT_BASED;
+
 /**
  * Loads classified segments from the database for a vert.x context.
  *
  * @author Armin Schnabel
  * @since 6.1.0
  */
 @SuppressWarnings("unused") // Part of the API
-public class ClassifiedSegmentRetriever {
+public class ClassifiedSegmentRetriever<T extends ClassifiedSegment> {
```

Comment:
This class can now load both, segments aggregated per-user and segments aggregated per-measurement (used as "history").